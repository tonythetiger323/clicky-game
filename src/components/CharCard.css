.card {
  cursor: pointer;
  margin: 0 auto;
  width: 256px;
  height: 256px;
  background-size: cover;
  background-clip: content-box;
  border-radius: 5px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
}

.card:hover {
  box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
}
