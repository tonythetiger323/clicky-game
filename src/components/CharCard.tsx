import React from "react";
import "./CharCard.css";

const CharCard = ({ id, name, image, handleSelected }: any) => (
  <div>
    <div
      className="card"
      key={id}
      data-id={id}
      style={{ backgroundImage: `url(${image})` }}
      onClick={handleSelected}
    />
  </div>
);

export default CharCard;
