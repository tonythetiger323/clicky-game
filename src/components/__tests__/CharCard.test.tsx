import React from "react";
import { shallow } from "enzyme";
import CharCard from "../CharCard";

describe("CharCard", () => {
  const charcard = shallow(<CharCard />);

  it("renders correctly", () => {
    expect(charcard).toMatchSnapshot();
  });
});
