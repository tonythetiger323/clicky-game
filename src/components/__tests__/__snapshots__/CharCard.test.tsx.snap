// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharCard renders correctly 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
</div>
`;
